src/cmd/cloud.js

```diff
@@ -260,17 +261,32 @@ module.exports = class CloudCommand extends CLICommandBase {
 			});
 	}
 
-	_getDownloadPath(deviceType, saveTo){
-		if (saveTo){
-			return saveTo;
+	_getDownloadPathForBin(deviceType, saveTo){
+		if (saveTo) {
+			// If saveTo has a .bin extension, use it. Otherwise,
+			// remove any existing extension and append .bin to the filename.
+			return (utilities.getFilenameExt(saveTo) === '.bin') ? saveTo : (utilities.filenameNoExt(saveTo) + '.bin');
```

Comment:
Made this modification because in the case of the following,
```
npm start -- compile /path/to/proj/folder/with/assets --saveTo myBundle.zip
```
the compiled binary is also ended up being called `myBundle.zip` so changing the logic to call app binary with .bin extension. I wonder this is backwards compatible.

Reply:
If you flip the logic around to 
```suggestion
			return (utilities.getFilenameExt(saveTo) === '.zip') ? (utilities.filenameNoExt(saveTo) + '.bin') : saveTo;
```
then it won't unintentionally change the behavior in case someone was doing `particle compile --saveTo filename_no_extension`